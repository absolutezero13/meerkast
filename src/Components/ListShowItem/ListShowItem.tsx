import React from "react";
import axios from "axios";
import Loader from "../Loader/Loader";
import { useState, useEffect } from "react";
import { Show } from "../../Redux/Reducers";
import ErrorPage from "../ErrorPage/ErrorPage";
const ListShowItem: React.FC<Show> = (props) => {
  const [link, setLink] = useState<string>("");
  const [loading, setLoading] = useState<boolean>(true);
  const [throwError, setErrorStatus] = useState<boolean>(false);
  useEffect(() => {
    axios
      .get(props.images["Poster Art"].url)
      .then((res) => {
        setLink(props.images["Poster Art"].url);
        setLoading(false);
      })
      .catch((error) => {
        setErrorStatus(true);
      });
  }, []);
  if (throwError) {
    return <ErrorPage />;
  }
  if (loading) {
    return <Loader />;
  }
  return (
    <div>
      <img alt={props.title} src={link} />
      <p>{props.title}</p>
    </div>
  );
};

export default ListShowItem;
