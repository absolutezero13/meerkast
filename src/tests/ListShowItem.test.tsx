import React from "react";
import { render } from "@testing-library/react";
import ListShowItem from "../Components/ListShowItem/ListShowItem";
import { testShowData } from "../fixtures/testShowData";

test("list show item matches snapshot", () => {
  const { container } = render(<ListShowItem {...testShowData[0]} />);
  expect(container).toMatchSnapshot();
});
