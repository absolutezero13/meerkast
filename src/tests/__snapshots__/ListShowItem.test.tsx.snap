// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list show item matches snapshot 1`] = `
<div>
  <div
    class="loading"
  >
    <div
      class="loader"
    />
  </div>
</div>
`;
